x-pack/plugins/security/public/views/management/edit_role/lib/validate_role.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class RoleValidator {
+  constructor(options = {}) {
+    this._shouldValidate = options.shouldValidate;
+  }
+
+  enableValidation() {
+    this._shouldValidate = true;
+  }
+
+  disableValidation() {
+    this._shouldValidate = false;
+  }
+
+  validateRoleName(role) {
+    if (!this._shouldValidate) return valid();
+
+    if (!role.name) {
+      return invalid(`Please provide a role name`);
+    }
+    if (role.name.length > 1024) {
+      return invalid(`Name must not exceed 1024 characters`);
+    }
+    if (!role.name.match(/^[a-zA-Z_][a-zA-Z0-9_@\-\$\.]*$/)) {
+      return invalid(`Name must begin with a letter or underscore and contain only letters, underscores, and numbers.`);
+    }
+    return valid();
+  }
+
+  validateIndexPrivileges(role) {
+    if (!this._shouldValidate) return valid();
+
+    if (Array.isArray(role.indices)) {
+      const areIndicesValid = role.indices
+        .map(this.validateIndexPrivilege.bind(this))
+        .find((result) => result.isInvalid) == null;
+
+      if (areIndicesValid) {
+        return valid();
+      }
+      return invalid();
+    }
+  }
+
+  validateIndexPrivilege(indexPrivilege) {
+    if (!this._shouldValidate) return valid();
+
+    if (indexPrivilege.names.length && !indexPrivilege.privileges.length) {
```

Comment:
That (to me) is expected. Blank index privileges are considered "placeholder" privileges, and they get stripped away before saving. This should be similar to how the old form worked. 
Once you specify an index, then you're required to add one or more privileges. Otherwise, you're allowed to leave it empty and we will ignore it.

Reply:
Gotcha, that's fine then if we're emulating how this used to be done, it felt different this time around. Ignore me on this one.